File: Leetcode/Weekly Contest 233/WC233-A.cpp
class Solution {
public:
	int maxAscendingSum(vector<int>& nums) {
		int res = nums[0], sum = nums[0];
		for (int s = 1; s < nums.size(); s++)
		{
			if (nums[s - 1] >= nums[s])
				sum = nums[s];
			else
				sum += nums[s];
			res = max(res, sum);
		}
		return res;
	}
};

File: Leetcode/Weekly Contest 233/WC233-A.py
class Solution(object):
    def maxAscendingSum(self, nums):
        res, sum = nums[0], nums[0]
        for i in range(1, len(nums)):
            if nums[i-1] >= nums[i]:
                sum = nums[i]
            else:
                sum += nums[i]
            res = max(res, sum)
        return res